src/Microsoft.SqlTools.ResourceProvider.Core/Authentication/IAccountManager.cs

```diff
@@ -0,0 +1,57 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.SqlTools.ResourceProvider.Core.Extensibility;
+
+namespace Microsoft.SqlTools.ResourceProvider.Core.Authentication
+{
+    /// <summary>
+    /// An account manager has the information of currently logged in user and can authenticate the user   
+    /// Implementing classes must add a <see cref="ExportableAttribute" />
+    /// to the class in order to be found by the extension manager,
+    /// and to define the type and category supported
+    /// </summary>  
+    public interface IAccountManager : IExportable
+    {
+        /// <summary>
+        /// Returns true is user needs reauthentication
+        /// </summary>
+        Task<bool> GetUserNeedsReauthenticationAsync();
+
+        /// <summary>
+        /// Authenticates the user
+        /// </summary>
+        Task<IUserAccount> AuthenticateAsync();
+
+        /// <summary>
+        /// Prompt the login dialog to login to a new use that has not been cached
+        /// </summary>
+        Task<IUserAccount> AddUserAccountAsync();
+
+        /// <summary>
+        /// Set the current loaged in user in the cache
+        /// </summary>
+        Task<IUserAccount> SetCurrentAccountAsync(object account);
+
+        /// <summary>
+        /// Returns the current logged in user
+        /// </summary>
+        Task<IUserAccount> GetCurrentAccountAsync();
+
+        /// <summary>
+        /// Returns true if the API supports a login control
+        /// </summary>
+        bool HasLoginDialog
```

Comment:
>HasLoginDialog [](start = 13, length = 14)

I assume a lot of the code is coming over as-is, so I won't comment further on code that may not make sense in the tools service context.

Reply:
Yes. Keeping most of this since it'll make cross-porting a bit easier in the long term which (hopefully) we want to keep as an option.